### PR TITLE
Decouple test suit

### PR DESCRIPTION
**Pull request by shehand**
## Purpose
This PR is to decouple the test suit in ballerina. And also this changes would remove the dependency of bLangPackage and TesterinaClassLoader dependency from the test suit while introducing TestMetaData (to store the details that are needed to run the test suit) and TestJsonData (store and retrieve data from a JSON file that are needed to init the test suit) entities.

Fixes https://github.com/ballerina-platform/ballerina-lang/issues/19755
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/19756

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java [hunk 1]
```diff
@@ -452,26 +451,26 @@ private static void resolveFunctions(TestSuite suite) {
                 test.setDataProviderFunction(functions.stream().filter(e -> e.getName().equals(test.getDataProvider()
                 )).findFirst().map(func -> {
                     // TODO these validations are not working properly with the latest refactoring
-                    if (func.getbFunction().getReturnTypeNode() != null) {
```
Comment (warunalakshitha) [inline misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: Why we comment out ordering tests

Comment (shehand) [reply misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/core/BTestRunner.java]: This resolveFunction method gets bFunction from TesterinaFunction class. The functionality of the if clause already commented and only else condition was there. It is also to return an exception. The idea was to comment on them to remove the dependency of bFunction from TesterinaFunctions.

**Comment by rasika** [comment]
@shehand build is failing due to packerina test failure.

**Comment by anupama-pathirage** [comment]
@hevayo @rasika @warunalakshitha Since @shehand is unavailable shall we update this PR and get this merged?

**Comment by warunalakshitha** [comment]
Closing this pr. We will merge the changes with https://github.com/ballerina-platform/ballerina-lang/pull/21150

**Pull request closed without merging**